CMakeLists.txt

```diff
@@ -620,13 +632,36 @@ if (USE_NCNN)
   include_directories(${NCNN_INC_DIR})
 endif()
 
+if (USE_TENSORRT)
+  find_package(Protobuf REQUIRED)
+  include_directories(${Protobuf_INCLUDE_DIRS})
+  include_directories(${CMAKE_CURRENT_BINARY_DIR})
+  if (JETSON)
+    set(TENSORRT_LIB_DEPS /usr/lib/aarch64-linux-gnu/libprotobuf.so -lnvinfer -lnvparsers -lnvinfer_plugin -lnvonnxparser)
+    set(TENSORRT_LIB_DIR /usr/lib/aarch64-linux-gnu)
+    set(TENSORRT_INC_DIR /usr/include/aarch64-linux-gnu)
+    include_directories(${TENSORRT_INC_DIR})    
+  elseif(DEFINED TENSORRT_DIR)
+    set(TENSORRT_LIB_DEPS /usr/lib/x86_64-linux-gnu/libprotobuf.so -lnvinfer -lnvparsers -lnvinfer_plugin -lnvonnxparser)
+    set(TENSORRT_LIB_DIR ${TENSORRT_DIR}/lib)
+    set(TENSORRT_INC_DIR ${TENSORRT_DIR}/include)
+    include_directories(${TENSORRT_INC_DIR})
+  else()
+    set(TENSORRT_LIB_DEPS /usr/lib/x86_64-linux-gnu/libprotobuf.so -lnvinfer -lnvparsers -lnvinfer_plugin -lnvonnxparser)
+    set(TENSORRT_LIB_DIR /usr/lib/x86_64-linux-gnu)
+    set(TENSORRT_INC_DIR /usr/include/x86_64-linux-gnu)
+    include_directories(${TENSORRT_INC_DIR})    
+  endif()
+endif()
+
 # main library, main & tests
 include_directories ("${PROJECT_SOURCE_DIR}/src")
 add_subdirectory (src)
 
 # Flags used by the subdirectories
-set(COMMON_INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
+set(COMMON_INCLUDE_DIRS ${Boost_INCLUDE_DIRS} ${CUDA_INCLUDE_DIRS})
```

Comment:
Maybe we want to add `CUDA_INCLUDE_DIRS` inside a `CUDA_FOUND` conditional ? I guess the assumption here is that `CUDA_INCLUDE_DIRS` is empty otherwise, but this might make funny things when CUDA is found and that a CPU_ONLY build is requested.

Reply:
ok, done